bgpd/bgp_nexthop.c

```diff
@@ -430,6 +430,52 @@ int bgp_multiaccess_check_v4(struct in_addr nexthop, struct peer *peer)
 	return (ret);
 }
 
+int bgp_subgrp_multiaccess_check_v4(struct in_addr nexthop,
+				    struct update_subgroup *subgrp)
+{
+	struct bgp_node *rn1, *rn2;
+	struct peer_af *paf;
+	struct prefix p, np;
+	bool first = true;
+	struct bgp *bgp = NULL;
+
+	np.family = AF_INET;
+	np.prefixlen = IPV4_MAX_BITLEN;
+	np.u.prefix4 = nexthop;
+
+	p.family = AF_INET;
+	p.prefixlen = IPV4_MAX_BITLEN;
+
+	rn1 = rn2 = NULL;
+	SUBGRP_FOREACH_PEER(subgrp, paf) {
+		if (first) {
```

Comment:
Couldn't we move this block to just before the FOREACH loop? If possible that would make the code a bit easier to read, not to mention the micro optimization. It seems to me that `bgp` can be obtained with `SUBGRP_INST(subgrp)`.

Reply:
I missed SUBGRP_INST, will move